ci/all_tests.sh

```diff
@@ -35,7 +38,7 @@ for roc_file in $EXAMPLES_DIR*.roc; do
         continue
     fi
 
-    $ROC build $roc_file $ROC_BUILD_FLAGS
+    $ROC build --linker=legacy --prebuilt-platform $roc_file $ROC_BUILD_FLAGS
```

Comment:
Should we use the default/surgical linker again here or not @lukewilliamboswell?

Reply:
I think we can remove this flag. We are building the surgical linker binaries now so that should be fine. 

We can also remove the `--prebuilt-platform` flag when https://github.com/roc-lang/roc/pull/6859 lands.